src/plot/voronoi.js

```diff
@@ -0,0 +1,53 @@
+import React, {PropTypes} from 'react';
+import {voronoi} from 'd3-voronoi';
+
+const NOOP = f => f;
+
+function Voronoi({className, extent, nodes, onBlur, onClick, onHover, x, y}) {
+  // Create a voronoi with each node center points
+  const voronoiInstance = voronoi()
+    .x(x)
+    .y(y)
+    .extent(extent);
+
+  return (
+    <g className={`${className} rv-voronoi`}>
+      {voronoiInstance.polygons(nodes).map((d, i) => (
+        <path
+          d={`M${d.join('L')}Z`}
+          onClick={() => onClick(d.data)}
+          onMouseOver={() => onHover(d.data)}
+          onMouseOut={() => onBlur(d.data)}
+          fill="none"
+          style={{pointerEvents: 'all'}}
+          key={i} />
+      ))}
+    </g>
+  );
+}
+
+Voronoi.requiresSVG = true;
```

Comment:
it's a react-vis property that causes it to be registered into the xy-plot svg rather than the xy-plot regular dom stuff